WhereInTheWorld/WhereInTheWorld/MapViewController.swift

```diff
@@ -0,0 +1,135 @@
+//
+//  ViewController.swift
+//  WhereInTheWorld
+//
+//  Created by Max Zvyagin on 2/10/20.
+//  Copyright © 2020 Max Zvyagin. All rights reserved.
+//
+
+import UIKit
+import MapKit
+
+class MapViewController: UIViewController, PlacesFavoriteDelegate {
+    
+    @IBOutlet var mapView: MKMapView!{
+        didSet{mapView.delegate = self as MKMapViewDelegate}
+    }
+    @IBOutlet var viewLocation: UILabel!
+    @IBOutlet var viewDescription: UILabel!
+    @IBOutlet var favButton: UIButton!
+    
+    
+    @IBOutlet var starButton: UIButton!
+    
+    var currentPlace: Place? = nil
+    
+    var allPlaces = [Place]()
+    
+    let defaults = UserDefaults.standard
+    
+    @IBAction func starButtonTap(_ sender: UIButton) {
+        // if it is currently selected, remove from favorites
+        if DataManager.sharedInstance.favorites.contains(currentPlace?.name ?? "blank"){
+            DataManager.sharedInstance.deleteFavorite(place: currentPlace?.name ?? "blank")
+        }
+        // need to have this in case someone hits the star when it's on a pin with multiple things
+        else if currentPlace?.name == nil{
+            return
+        }
+        // else add to the favorites
+        else{
+            DataManager.sharedInstance.favorites.append(currentPlace!.name!)
+        }
+        // edit so button image will change
+        sender.isSelected = !sender.isSelected
+        // add to user defaults holding map location in this function
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view.
+        // set map view properties
+        mapView.showsCompass = false
+        mapView.pointOfInterestFilter = .excludingAll
+        // set star button images
+        starButton.setImage(UIImage(systemName:"star.fill"), for: .selected)
+        starButton.setImage(UIImage(systemName:"star"), for: .normal)
+        // load in the data from plist
+        let places = DataManager.sharedInstance.loadAnnotationFromPlist(filename:"Data")
+        self.allPlaces = places // save the list of all places
+        self.mapView.addAnnotations(places)
+        // edit favorites button
+        self.favButton.layer.cornerRadius=10
+    }
+    
+    // not sure if this is right with the table view segue
+    override func viewWillAppear(_ animated: Bool) {
```

Comment:
This is another lifecycle method. It should live at the top of your file, not after an IBAction func.